src/main/java/org/jabref/logic/citationkeypattern/CitationKeyGenerator.java

```diff
@@ -102,10 +110,15 @@ public static String cleanKey(String key, String unwantedCharacters) {
      */
     public String generateKey(BibEntry entry) {
         Objects.requireNonNull(entry);
+
+        LatexCleanupFormatter formatter = new LatexCleanupFormatter();
+        String formattedTitle = formatter.format(entry.getField(StandardField.TITLE).orElse(""));
+        entry.setField(StandardField.TITLE, formattedTitle);
```

Comment:
This is wrong! The entry MUST NOT be modified.

I think, you can just delete these two lines.